bn_mp_set_i32.c

```diff
@@ -0,0 +1,15 @@
+#include "tommath_private.h"
+#ifdef BN_MP_SET_I32_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+
+
+/* Set the least signficant 32 bit of "a" to b, signed */
+void mp_set_i32(mp_int *a, int32_t b)
+{
+   mp_set_u32(a, (b < 0) ? -(uint32_t)b : (uint32_t)b);
```

Comment:
Since negation of an unsigned doesn't make much sense (see previous comment), better do negation first, then the (uint32_t) type-cast

Reply:
To elaborate, this would introduce undefined behavior for -0x80000000, resulting in signed integer overflow.